Update product-versioning.md (#1387)
bonitasoft/bonita-doc: md/product-versioning.md

## Product Versions

After years of semantic versioning, Bonita now follows the calendar versioning: YYYY.#-mmdd, since version 2021.1:
* YYYY is the year of the release

> * \# is the release number. Two versions are released every year.

Reviewer: why an anti-slash?

Author: # is used for title \ is an escape character

> * mmdd is the month and day of the release. A new maintenance version of the released number is made available every month. For example the february version will be 2021.1-0205.Keep in mind that all maintenance version are cross-compatible (see the Technical Id chapter)

Reviewer: I do not get "of the released number".

> * mmdd is the month and day of the release. A new maintenance version of the released number is made available every month. For example the february version will be 2021.1-0205.Keep in mind that all maintenance version are cross-compatible (see the Technical Id chapter)

Reviewer: ```suggestion
* mmdd is the month and day of the release. A new maintenance version is made available every month. For example, the February version will be 2021.1-0205 as it will be released on the 5th. Keep in mind that all maintenance versions are cross-compatible (see the Technical Id chapter).
```

## Technical Id

> A technical Id also exists and it follows the usual semantic versioning X.Y.z:

Reviewer: ```suggestion
A technical Id also exists for Bonita components, and it follows the usual semantic versioning X.Y.Z:
```

* X is the major version number. It changes when non-backward compatible changes are implemented.
* Y is the minor version number. It version changes when backward compatible features are added. A minor version is released every six months to improve continuously Bonita.

> * Z is the maintenance version number. The maintenance versions contain fixes. Maintenance versions are released every month between two minor versions for the current minor version. These versions are cross-compatible by default for the same minor version (for the runtime and Studio).

Reviewer: ```suggestion
* Z is the maintenance version number. The maintenance versions contain fixes. Maintenance versions are released every month between two Bonita YYYY.# versions. These versions are cross-compatible by default for the same YYYY.# version (for Bonita Runtime and Bonita Studio).
```

> For example, upgrading from a 7.12.2 to a 7.12.5 will not require any migration. However, upgrades from a 7.11.5 to a 7.12.2  or to  8.0.1 will require a migration.

Reviewer: ```suggestion
For example, upgrading from a 2021.1-0205 to a 2021.1-0401 will not require any migration. However, upgrades from a 7.11.5 to a 2021.1-0205 will require a migration.
```

## Product Versions

> Bonita follows calendar versioning: YYYY.n:

Reviewer: ```suggestion
After years of semantic versioning, Bonita now follows calendar versioning: YYYY.n, since version 2021.1:
```

* YYYY year of the release
* n is the release number. Two versions are released every year.

### Technical Id

A technical Id also exists and it follows the usual semantic versioning X.Y.z:
* X is the major version number. It changes when non-backward compatible changes are implemented.
* Y is the minor version number. It version changes when backward compatible features are added. A minor version is released every six months to improve continuously Bonita.
* Z is the maintenance version number. The maintenance versions contain fixes. Maintenance versions are released every month between two minor versions for the current minor version. These versions are cross-compatible by default for the same minor version.

> 

Reviewer: There we should introduce the maintenance version new numbering: YYYY.#-mmdd

> For example, upgrading from a 7.12.2 to a 7.12.5 will not require any migration. However, migrating from a 7.11.5 to a 7.12.2  or to  8.0.1 will require a migration.

Reviewer: You can also add than since 2021.1, Bonita Studio supports projects that are created in any maintenance version for a given minor version.
